processor/src/main/java/org/mapstruct/ap/internal/model/beanmapping/SourceReference.java

```diff
@@ -281,13 +281,22 @@ private void reportErrorOnNoMatch( Parameter parameter, String[] propertyNames,
                     propertyNames[notFoundPropertyIndex],
                     sourceType.getPropertyReadAccessors().keySet()
                 );
-                List<String> elements = new ArrayList<>(
-                    Arrays.asList( propertyNames ).subList( 0, notFoundPropertyIndex )
-                );
-                elements.add( mostSimilarWord );
-                reportMappingError(
-                    Message.PROPERTYMAPPING_INVALID_PROPERTY_NAME, sourceName, Strings.join( elements, "." )
-                );
+                if ( mostSimilarWord != null ) {
```

Comment:
I guess that  reeds a lot better when you do it like this:
```java
if ( !sourceType.getPropertyReadAccessors().keySet().isEmpty() ) {

    String mostSimilarWord = Strings.getMostSimilarWord(...  ) 
    ... 
}
else { 
  ...
} 
```

Reply:
You are right. I'll change it to this